go/merkletree2/position.go

```diff
@@ -127,6 +141,33 @@ func (t *Config) getDeepestPositionForKey(k Key) (*Position, error) {
 	return &p, nil
 }
 
+// Returns the lexicographically first key which could be found at any children
+// of position p in the tree
+func (t *Config) getMinKey(p *Position) Key {
+	var min big.Int
+	min.Set((*big.Int)(p))
+	n := uint(t.KeysByteLength*8 + 1 - min.BitLen())
+	min.Lsh(&min, n)
+	return min.Bytes()[1:]
+}
+
+func (t *Config) GetKeyIntervalUnderPosition(p *Position) (minKey, maxKey Key) {
+	var min, max big.Int
+
+	min.Set((*big.Int)(p))
+	n := uint(t.KeysByteLength*8 + 1 - min.BitLen())
+	min.Lsh(&min, n)
+	minKey = min.Bytes()[1:]
```

Comment:
this is same as getMinKey?

Reply:
It is, but I am reusing the min object to compute the max, which is why I do not call the getMin function directly.